src/list.js

```diff
@@ -130,7 +130,7 @@ class TypeInferedList extends BaseImmutableList {
   }
 
   get(index, notSetValue) {
-    return this[$store] ? this[$store].get(index, notSetValue) :
+    return this[$store] ? this[$store].get(parseInt(index), notSetValue) :
```

Comment:
We should keep it as it was and match Immutable.js - this is actually an intended change, see:

https://github.com/facebook/immutable-js/commit/7acd81b2bf0d9ecc511a08559320d9cc83407eef

The original test cases were based on the immutable.js test cases